opennlp-tools/src/main/java/opennlp/tools/dictionary/Dictionary.java

```diff
@@ -343,9 +343,41 @@ public boolean contains(Object obj) {
           result = entrySet.contains(new StringListWrapper(new StringList(str)));
 
         }
-
         return result;
       }
+
+      @Override
+      public boolean equals(Object o) {
+        if (! (o instanceof Set)) {
+          return false;
+        } else {
+          Set<String> toCheck = (Set<String>) o;
+          if (entrySet.size() != toCheck.size()) {
+            return false;
+          } else {
+            boolean isEqual = true; // will flip if one element is not equal
+            Iterator<StringListWrapper> entrySetIter = entrySet.iterator();
+            Iterator<String> toCheckIter = toCheck.iterator();
+            for (int i = 0; i < entrySet.size(); i++) {
+              StringListWrapper entry = entrySetIter.next();
+              if (isCaseSensitive) {
+                isEqual = entry.stringList.equals(new StringList(toCheckIter.next()));
+              } else {
+                isEqual = entry.stringList.compareToIgnoreCase(new StringList(toCheckIter.next()));
+              }
+              if (!isEqual) {
+                break;
+              }
+            }
+            return isEqual;
+          }
+        }
+      }
```

Comment:
I think this one has the suggestions above, and doesn't need the `isEqual` variable.

```suggestion
@Override
      public boolean equals(Object o) {
        if (! (o instanceof Set)) {
          return false;
        }
        Set<String> toCheck = (Set<String>) o;
        if (entrySet.size() != toCheck.size()) {
          return false;
        }
        Iterator<String> toCheckIter = toCheck.iterator();
        for (StringListWrapper entry : entrySet) {
          if (isCaseSensitive) {
            if (!entry.stringList.equals(new StringList(toCheckIter.next()))) {
              return false;
            }
          } else {
            if (!entry.stringList.compareToIgnoreCase(new StringList(toCheckIter.next()))) {
              return false;
            }
          }
        }
        return true;
      }
```

Reply:
@kinow Will check/test it.

Reply:
LGTM

Reply:
Adapted and pushed.